### DISCUSSION TITLE
Support denote file-naming scheme

### DISCUSSION THREAD

**Discussion by TristanCacqueray** [srid/emanote#551] [Ideas]
Hello, I am evaluating [denote](https://protesilaos.com/emacs/denote) and I am looking for using it with emanote.

In short, denote uses the following file naming convention: `DATE--TITLE__KEYWORDS.EXT`, the `DATE` acting like a uid. Here is an example file named `20240926T193355--this-is-a-test__denote_emanote.md`:

```markdown
---
title:      "This is a test"
date:       2024-09-26T19:33:55-04:00
tags:       ["denote", "emanote"]
identifier: "20240926T193355"
---

A link to [another-note](denote:20240919T145008).
```

> The notes can be saved in sub-directories, and the naming scheme also support Luhmann-style signatures between the date and title like this: `DATE==sig--title...`.

Looking at the `Emanote.Model.Link` modules, it doesn't seem too complicated to handle the `denote:` linking scheme, so my request is more about the route slug. I would like to render the html as `2024-09-26_this-is-a-test`, e.g. by removing the time and tags (when the file name match the denote convention). Where should I implement this renaming and how to handle conflicts?

Does that sounds like a good idea to you (cc @srid)?

Cheers,
-Tristan

**Comment by srid** [comment]
This is basically like #552 but with different scheme logic.

How about writing a general mechanism that can polymorphically support many such schemes like this? Thus, there will be less of hardcoded logic in code (as those logic would be decoupled into the concept of schemes).

**Reply by srid** [reply]
> Where should I implement this renaming and how to handle conflicts?

That reminds me, I should find time to do #556 (especially to get back to hacking on Haskell since it has been a while I touched Haskell!)

**Reply by TristanCacqueray** [reply]
It's a bit more than #552 since we would need to use the timestamp as a linkable id, like by adding DenoteIdentifier to `NoteIxs`.

For conflicts, I don't expect it to be an issue in practice, e.g. if we format the note as `YYYY-MM-DD-titleslug`, then as long as you don't create two notes with the same title on the same day, then the routes won't conflict.

Denote looks like a great system, and I think emanote would do a good job at publishing the notes since it already has the base machinery for backlinks and tags.
